lib/history.js

```diff
@@ -51,6 +55,9 @@ function runInvalidator () {
   for (var i = 0; i < pids.length; i++) {
     if (expiration[pids[i]] < now) {
       size--
+      if (history[pids[i]].fd) {
```

Comment:
I would have added an `onExpire` callback to the add function instead of doing this.
But that absolutely has the advantage of using less memory.